server/sonar-main/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -165,7 +178,15 @@ private ProcessBuilder create(EsCommand esCommand) {
     commands.addAll(javaCommand.getJvmOptions().getAll());
     commands.addAll(buildClasspath(javaCommand));
     commands.add(javaCommand.getClassName());
-    commands.add(buildPropertiesFile(javaCommand).getAbsolutePath());
+    if (javaCommand.getReadsArgumentsFromFile()) {
+      commands.add(buildPropertiesFile(javaCommand).getAbsolutePath());
+    } else {
+      javaCommand.getArguments().forEach((key, value) -> {
+        if (value != null && !value.isEmpty()) {
+          commands.add("-E" + key + "=" + value);//FIXME make this safer (or remove it, if not needed, because there is an elasticsearch.yml anyways
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)